[#1907] Initial working version of wikipedia plugin

cadecairos opened the pull request:
Test in basic template

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d4ab7d37b308714/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/d4ab7d37b308714/output.txt

Total script time: 1.46 mins

Server is running on http://pr740.butyr.org/templates/basic/

Commit:
[#1907] Initial working version of wikipedia plugin

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/489c5b3e8a479ae/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/489c5b3e8a479ae/output.txt

Total script time: 0.76 mins

Server is running on http://pr740.butyr.org/templates/basic/

Commit:
[#1907] Redesign the plugin

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f5a9229f496e86e/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/f5a9229f496e86e/output.txt

Total script time: 1.02 mins

Server is running on http://pr740.butyr.org/templates/basic/

Commit:
[#1907] Modify config files for new wikipedia plugin

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/9e9c66cddc00574/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/9e9c66cddc00574/output.txt

Total script time: 0.93 mins

Server is running on http://pr740.butyr.org/templates/basic/

Commit:
[#1907] CSS fix

k88hudson commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/40cbd7e0b77e193/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/40cbd7e0b77e193/output.txt

Total script time: 0.73 mins

Server is running on http://pr740.butyr.org/templates/basic/

Commit:
[#1907] Style fixes, made wiki plugin resizable in basic, and added all supported wikipedia languages to the manifest, default is English.

cadecairos commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/fe07e90a283b1f0/output.txt

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/70879873d4802c4/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/fe07e90a283b1f0/output.txt

Total script time: 0.97 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/70879873d4802c4/output.txt

Total script time: 0.90 mins

Server is running on http://pr740.butyr.org/templates/basic/
Run the Test Suite: http://pr740.butyr.org/test/

Commit:
[#1907] Re-structure wikipedia plugin for clarity, add values attribute to Language options

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4fcd6321f76f6f5/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/4fcd6321f76f6f5/output.txt

Total script time: 0.80 mins

Server is running on http://pr740.butyr.org/templates/basic/
Run the Test Suite: http://pr740.butyr.org/test/

mjschranz commented on src/default-config.json:
```diff
@@ -31,10 +31,6 @@
-      },
-      {
-        "type": "wikipedia",
-        "path": "{{baseDir}}external/popcorn-js/plugins/wikipedia/popcorn.wikipedia.js"
```
Why remove this from the default config? Our basic template will overwrite this anyway.

cadecairos replied:
Is there a reason to keep it?

humphd replied:
Please modify the path so that it points to your plugin.  Our build system uses this info to create the complete popcorn.js file.  Should test it too `node make buttered-popcorn`.

mjschranz replied:
The problem with including these plugins in the default config right now is that any templates using them would need the CSS pages and various editor helpers included with them.

I think it's best to leave the default config using the base plugins we have in Popcorn.js and leave it at that.

mjschranz commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+      _container.style.width = ( options.width || _manifestOpts.width[ "default" ] ) + "%";
+      _container.style.height = ( options.height || _manifestOpts.height[ "default" ] ) + "%";
+      _container.style.top = ( options.top || _manifestOpts.top[ "default" ] ) + "%";
+      _container.style.left = ( options.left || _manifestOpts.left[ "default" ] ) + "%";
```
What if the user decides to drag this beyond the video? I'm fairly certain it's something we don't want to happen. Should consider stealing the normalize function we use in other plugins like popup.

cadecairos replied:
one-lined that normalize function: 

```
function normalize( value, minWidth, maxWidth ) {
    return Math.max( Math.min( value || 0, maxWidth ), minWidth );
}
```

mjschranz commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+      _container.style.left = ( options.left || _manifestOpts.left[ "default" ] ) + "%";
+
+      _titleDiv = create( "div" );
+      _titleDiv.classList.add( "wikipedia-title");
```
space right bracket

mjschranz commented:
/botio check

Commit:
[#1907]
1. Add normalize function to set max and min values for top, left, height and width
2. Style and lint fixes

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+        options.lang = "en";
+      }
+
+      window[ "wikiCallback" + _guid ]  = function ( data ) {
```
Popcorn.guid let's you pass a prefix string, so your guid could be "wikiCallback.......".

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+      window[ "wikiCallback" + _guid ]  = function ( data ) {
+
+        if ( data.error ) {
+          _titleTextArea.innerHTML = "Uh oh....";
```
Can we do better than this error message?

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+        options._link = "//" + options.lang + ".wikipedia.org/wiki/" + _title;
+
+        // gets the mobile format, so that we don't load unwanted images when the respose is turned into a documentFragment
+        Popcorn.getScript( _href + "api.php?action=parse&prop=text&redirects&page=" +
```
you probably want to uriencode the title and other strings, so you don't get into XSS trouble.

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+    },
+
+    end: function( event, options ){
+
```
remove first blank line if not necessary (it's not here).

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,274 @@
+        elem: "input",
+        type: "text",
+        label: "Article Link/Title",
+        "default": "London_2012_Olympics"
```
We should get something else in here, this won't be topical in 6 months.

Commit:
[#1907]
1. Review Fixes
2. Add icon for plugin

cadecairos commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/e1b413ade613346/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/e1b413ade613346/output.txt

Total script time: 0.75 mins

Server is running on http://pr740.butyr.org/templates/basic/
Run the Test Suite: http://pr740.butyr.org/test/

humphd commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/5e51a2b3cd99993/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/5e51a2b3cd99993/output.txt

Total script time: 1.21 mins

humphd commented on templates/assets/editors/wikipedia.js:
```diff
@@ -0,0 +1,18 @@
+EditorHelper.addPlugin( "wikipedia", function( trackEvent ) {
+  var _container,
+      _popcornOptions,
+      media;
```
Either use _ or don't, but don't mix.

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,272 @@
+"use strict";
```
We haven't really gotten into using strict throughout our code.  Any reason why you're using here?

cadecairos replied:
I was using it to help catch issues easily in my work. it can be pulled out.

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,272 @@
+  function getFragment( inputString ) {
+    //grabbed from butter util methods
+    var range = document.createRange(),
+
```
remove blank line

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,272 @@
+        options.lang = "en";
+      }
+
+      window[ _guid ]  = function ( data ) {
```
extra space after closing ]

humphd commented on templates/assets/plugins/wikipedia/popcorn.wikipedia.js:
```diff
@@ -0,0 +1,272 @@
+          return;
+        }
+
+        var responseFragment = getFragment( "<div>" + data.parse.text + "</div>" ),
```
You might want to sanitize or escape this text, since you're putting it directly into HTML.

Commit:
[#1907] Review fixes for Wikipedia plugin

jbuck commented:
Staged on master: https://github.com/mozilla/butter/commit/1aa6d8ca84a45dfb8c6f64ab0c11b88a0cb8426b
